Feature/testing component Bryan Chamorro test

8rch opened the pull request:
Now the code efficiently loads and compiles the JSON schema, and then uses it to validate the response JSON in the shouldValidateJsonSchema() test method.
important 
To run the test, you can use your IDE's test runner or use the Maven command mvn test.
I added an exception handler method to handle any exceptions and return an appropriate response with an error message.

8rch committed:
adding test files and parcial build

8rch committed:
adding test files

8rch committed:
adding test files and parcial build

8rch committed:
new file:   mvnRes/.gitignore
	new file:   mvnRes/pom.xml
	new file:   mvnRes/src/main/java/org/example/Main.java
	new file:   mvnRes/src/test/java/GameResponse.java
	new file:   mvnRes/src/test/java/SchemaValidator.java
	modified:   pom.xml
	modified:   src/main/java/fun/Game.java
	modified:   src/main/java/fun/GameController.java
	modified:   src/test/java/fun/GameTests.java
	deleted:    test/out/production/test/Main.class
	deleted:    test/pom.xml
	modified:   test/test.iml

echudotdash commented:
as mentioned in README, can you do this excise by using REST Assured Lib ? https://rest-assured.io/

8rch committed:
REST Assured Lib used in GameController

8rch commented:
I updated the code, I replaced the TestRestTemplate and ResponseEntity with REST Assured's fluent API for making HTTP requests and performing assertions.

8rch committed:
pom dependency updated
